codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/middleware/MiddlewareBuilder.java

```diff
@@ -230,55 +230,28 @@ public void addDefaultMiddleware(GenerationContext context) {
                         "Enable response stubbing for testing. See {Hearth::ClientStubs#stub_responses}.")
                 .build();
 
-        ClientConfig maxAttempts = (new ClientConfig.Builder())
-                .name("max_attempts")
-                .type("Integer")
-                .defaultValue("3")
+        ClientConfig retryStrategy = (new ClientConfig.Builder())
+                .name("retry_strategy")
+                .type("Hearth::Retry::Strategy")
+                .documentationDefaultValue("Hearth::Retry::Standard.new")
+                .defaultValue("Hearth::Retry::Standard.new")
                 .documentation(
-                        "An integer representing the maximum number of attempts that will be made for a "
-                                + "single request, including the initial attempt."
-                )
-                .build();
-
-        ClientConfig retryMode = (new ClientConfig.Builder())
-                .name("retry_mode")
-                .type("String")
-                .defaultValue("'standard'")
-                .documentation(
-                        "Specifies which retry algorithm to use. Values are: \n"
-                                +
-                                " * `standard` - A standardized set of retry rules across the AWS SDKs. "
-                                +
-                                "This includes support"
-                                +
-                                " for retry quotas, which limit the number of unsuccessful retries a client can make.\n"
-                                + " * `adaptive` - An experimental retry mode that includes all the"
+                        "Specifies which retry strategu to use. Values are: \n"
+                                + " * `Retry::Standard` - A standardized set of retry rules across the AWS SDKs. "
```

Comment:
Its a little confusing with "the values are" since these need to be instantiated.  Maybe "values are instances of: ".  And maybe mention that retry configuration such as max retries must be configured on the retry strategy.